lib/decidim/spam_signal/spammer_signaler/spammer_signaler.rb

```diff
@@ -0,0 +1,114 @@
+require 'uri'
+
+module Decidim
+  module SpamSignal
+    class AppSpamSignal
+
+      def run!
+        @user_bot_email = ENV.fetch("USER_BOT_EMAIL", nil)
+        @uncheckable_email_domain = ENV.fetch("UNCHECKABLE_EMAIL_DOMAIN", nil)
+        @user_bot = take_or_create_user_bot
+        @regex = /#{take_regex}/i
+        
+        users_to_check.each do |user|
+          unless user_signaled?(user) || user.blocked?
+            signal_spammer(user) if user_about_uri?(user) && user_about_text?(user)
+          end
+        end
+      end
+
+      private 
+        def take_or_create_user_bot
+            
+          raise "Organization doesn't exists" unless @organization = Decidim::Organization.first
```

Comment:
I see an immediate optimisation here: 

```
   raise "Organization doesn't exists" unless @organization = Decidim::Organization.first
```

A cleaner way would be to make an accessor: 
```
def organization
    Decidim::Organization.first! # will raise NotFound if not exists — note the "!"
end
```